addon/edit/continuelist.js

```diff
@@ -20,7 +20,23 @@
     var ranges = cm.listSelections(), replacements = [];
     for (var i = 0; i < ranges.length; i++) {
       var pos = ranges[i].head;
-      var eolState = cm.getStateAfter(pos.line);
+
+      // Walk the list of inner modes/states until we find a Markdown one
+      // If we don't find one, fall back to normal newlineAndIndent
+      var mode = cm.getMode(), eolState = cm.getStateAfter(pos.line);
+      while (mode && mode.name !== "markdown") {
```

Comment:
You don't need the loop here—`CodeMirror.innerMode` is a helper that'll go down to the innermost mode for you. Then you can check whether it's markdown.